7-gce/gce/deploy.sh

```diff
@@ -140,9 +144,16 @@ gcloud compute instance-groups managed set-autoscaling \
 # [END set_autoscaling]
 
 # [START create_firewall]
-gcloud compute firewall-rules create default-allow-http-8080 \
+# Check if the firewall rule has been created in previous steps of the documentation
+if gcloud compute firewall-rules list --filter="name~'default-allow-http-8080'" \
+  --format="table(name)" | grep -q 'NAME'; then
+  echo "Firewall rule default-allow-http-8080 already exists."
+else
+  gcloud compute firewall-rules create default-allow-http-8080 \
     --allow tcp:8080 \
     --source-ranges 0.0.0.0/0 \
     --target-tags http-server \
-    --description "Allow port 8080 access to http-server"
+    --description "Allow port 8080 access to http-server" 
+fi
+
```

Comment:
Old script may fail if user has created a _default-allow-http-8080_ firewall rule in previous steps of the tutorial. Added a new command to create the firewall rule only when the rule has not already existed.